web/controllers/group_pair_controller.ex

```diff
@@ -11,7 +11,13 @@ defmodule Pairmotron.GroupPairController do
 
     group = Repo.get(Group, group_id)
     if authorized?(group, conn.assigns.current_user) do
-      pairs = PairMaker.fetch_or_gen(year, week, group_id)
+      pairs = case PairMaker.fetch_or_gen(year, week, group_id) do
+        {:error, pairs, message} ->
+          conn |> put_flash(:error, message)
```

Comment:
So if I'm not mistaken, this isn't actually going to result in the conn having that flash message down below, since it's not actually being returned by the case statement (pairs is). The conn outside of this case will be unaltered. Yay, Immutability!